File: Java/Concentus/src/org/concentus/SilkDecoderControl.java
/* Copyright (c) 2006-2011 Skype Limited. All Rights Reserved
   Ported to Java by Logan Stromberg

   Redistribution and use in source and binary forms, with or without
   modification, are permitted provided that the following conditions
   are met:

   - Redistributions of source code must retain the above copyright
   notice, this list of conditions and the following disclaimer.

   - Redistributions in binary form must reproduce the above copyright
   notice, this list of conditions and the following disclaimer in the
   documentation and/or other materials provided with the distribution.

   - Neither the name of Internet Society, IETF or IETF Trust, nor the
   names of specific contributors, may be used to endorse or promote
   products derived from this software without specific prior written
   permission.

   THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
   ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
   LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
   A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER
   OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
   EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
   PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
   PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
   LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
   NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
   SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 */
package org.concentus;

/// <summary>
/// Decoder control
/// </summary>
class SilkDecoderControl {

    /* Prediction and coding parameters */
    final int[] pitchL = new int[SilkConstants.MAX_NB_SUBFR];
    final int[] Gains_Q16 = new int[SilkConstants.MAX_NB_SUBFR];

    /* Holds interpolated and final coefficients */
    final short[][] PredCoef_Q12 = Arrays.InitTwoDimensionalArrayShort(2, SilkConstants.MAX_LPC_ORDER);
    final short[] LTPCoef_Q14 = new short[SilkConstants.LTP_ORDER * SilkConstants.MAX_NB_SUBFR];
    int LTP_scale_Q14 = 0;

    void Reset() {
        Arrays.MemSet(pitchL, 0, SilkConstants.MAX_NB_SUBFR);
        Arrays.MemSet(Gains_Q16, 0, SilkConstants.MAX_NB_SUBFR);
        Arrays.MemSet(PredCoef_Q12[0], (short) 0, SilkConstants.MAX_LPC_ORDER);
        Arrays.MemSet(PredCoef_Q12[1], (short) 0, SilkConstants.MAX_LPC_ORDER);
        Arrays.MemSet(LTPCoef_Q14, (short) 0, SilkConstants.LTP_ORDER * SilkConstants.MAX_NB_SUBFR);
        LTP_scale_Q14 = 0;
    }
}

File: Go/opus/silk_decoderControl.go
package opus

type SilkDecoderControl struct {
	pitchL        []int
	Gains_Q16     []int
	PredCoef_Q12  [][]int16
	LTPCoef_Q14   []int16
	LTP_scale_Q14 int
}

func NewSilkDecoderControl() *SilkDecoderControl {
	obj := &SilkDecoderControl{}
	obj.pitchL = make([]int, SilkConstants.MAX_NB_SUBFR)
	obj.Gains_Q16 = make([]int, SilkConstants.MAX_NB_SUBFR)
	obj.LTPCoef_Q14 = make([]int16, SilkConstants.LTP_ORDER*SilkConstants.MAX_NB_SUBFR)
	/* Holds interpolated and final coefficients */
	obj.PredCoef_Q12 = InitTwoDimensionalArrayShort(2, SilkConstants.MAX_LPC_ORDER)
	return obj
}
func (p *SilkDecoderControl) Reset() {
	MemSetLen(p.pitchL, 0, SilkConstants.MAX_NB_SUBFR)
	MemSetLen(p.Gains_Q16, 0, SilkConstants.MAX_NB_SUBFR)
	MemSetLen(p.PredCoef_Q12[0], 0, SilkConstants.MAX_LPC_ORDER)
	MemSetLen(p.PredCoef_Q12[1], 0, SilkConstants.MAX_LPC_ORDER)
	MemSetLen(p.LTPCoef_Q14, 0, SilkConstants.LTP_ORDER*SilkConstants.MAX_NB_SUBFR)
	p.LTP_scale_Q14 = 0
}
